pkg/mdformatter/mdgen/mdgen.go

```diff
@@ -83,10 +83,9 @@ func (t *genCodeBlockTransformer) TransformCodeBlock(ctx mdformatter.SourceConte
 		cmd.Stdout = &b
 		if err := cmd.Run(); err != nil {
 			expectedCode, _ := strconv.Atoi(infoStringAttr[infoStringKeyExitCode])
-			if exitErr, ok := err.(*exec.ExitError); ok && exitErr.ExitCode() == expectedCode {
-				return b.Bytes(), nil
+			if exitErr, ok := err.(*exec.ExitError); !ok || exitErr.ExitCode() != expectedCode {
```

Comment:
Maybe we should return a separate error if `!ok` and `exitErr.ExitCode() != expectedCode`?

Reply:
Thanks, I've updated the PR to differentiate these errors.